components/server/src/websocket/websocket-connection-manager.ts

```diff
@@ -376,31 +377,31 @@ class GitpodJsonRpcProxyFactory<T extends object> extends JsonRpcProxyFactory<T>
     protected async onRequest(method: string, ...args: any[]): Promise<any> {
         const span = TraceContext.startSpan(method, undefined);
         const userId = this.clientMetadata.userId;
-        const requestId = span.context().toTraceId();
-        return runWithLogContext(
-            "request",
+        const abortController = new AbortController();
+        const cancellationToken = args[args.length - 1];
+        if (CancellationToken.is(cancellationToken)) {
+            cancellationToken.onCancellationRequested(() => abortController.abort());
+        }
+        return runWithRequestContext(
             {
-                userId,
-                contextId: requestId,
-                method,
+                requestKind: "jsonrpc",
```

Comment:
Can we make requestKind specify in interface? In case we do it duplicate with different format like `json-rpc`

Reply:
Hm, ideally we should only have a single call per kind. What about this: Let's see if we can do without it, and otherwise add later?